src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -625,15 +636,12 @@ namespace cryptonote
         b.block = arg.b.block;
         b.txs = have_tx;
 
-        std::list<block_complete_entry> blocks;
-        blocks.push_back(b);
-        m_core.prepare_handle_incoming_blocks(blocks);
-          
         block_verification_context bvc = boost::value_initialized<block_verification_context>();
         m_core.handle_incoming_block(arg.b.block, bvc); // got block from handle_notify_new_block
         if (!m_core.cleanup_handle_incoming_blocks(true))
         {
           LOG_PRINT_CCONTEXT_L0("Failure in cleanup_handle_incoming_blocks");
+          m_core.cleanup_handle_incoming_blocks();
```

Comment:
Wait, you just invoked this one line 641, why does this need to be here again?

Reply:
Oops, fixed. Introduced because I had misread handle_incoming_tx/handle_incoming_block. That patch is undeeded.